src/test/java/com/nvidia/spark/rapids/jni/ParseURITest.java

```diff
@@ -175,11 +177,135 @@ void testPath(String[] testData) {
+      try (ColumnVector protocolResult = ParseURI.parseURIProtocol(invalidData, false);
+           ColumnVector expectedProtocol = ColumnVector.fromStrings(new String[]{"https", null, "https", null})) {
+        AssertUtils.assertColumnsAreEqual(expectedProtocol, protocolResult);
+      }
+      
+      // ANSI mode should throw exception when encountering invalid URLs
+      try {
+        try (ColumnVector protocolResult = ParseURI.parseURIProtocol(invalidData, true)) {
+          fail("Expected exception for invalid URLs in ANSI mode");
+        }
+      } catch (Exception e) {
+        assertNotNull(e, "Expected an exception to be thrown in ANSI mode for invalid URLs");
+      }
+    }
+  }
+
+  @Test
+  void parseURIMissingQueryParamTest() {
+    // Test the specific case where query parameter is missing
+    String[] testData = {
+        "https://secure.payment.com/process?amount=100&currency=USD",  // has 'amount'
+        "http://analytics.site.com/track?event=click&user=456",        // no 'amount'
+        "ftp://backup.server.com/files/data.csv"                       // no query at all
+    };
+    
+    // Test with non-ANSI mode - should return nulls for missing params
+    try (ColumnVector v0 = ColumnVector.fromStrings(testData);
+         ColumnVector result = ParseURI.parseURIQueryWithLiteral(v0, "amount", false)) {
+      String[] expected = {"100", null, null};
+      try (ColumnVector expectedCV = ColumnVector.fromStrings(expected)) {
+        AssertUtils.assertColumnsAreEqual(expectedCV, result);
+      }
+    }
+    
+    // Test with different missing parameter to make sure it also returns nulls
+    try (ColumnVector v0 = ColumnVector.fromStrings(testData);
+         ColumnVector result = ParseURI.parseURIQueryWithLiteral(v0, "nonexistent", false)) {
+      String[] expected = {null, null, null};
+      try (ColumnVector expectedCV = ColumnVector.fromStrings(expected)) {
+        AssertUtils.assertColumnsAreEqual(expectedCV, result);
+      }
+    }
+    
+    // Test with ANSI mode - should ALSO return nulls for missing params, not throw exceptions
+    try (ColumnVector v0 = ColumnVector.fromStrings(testData);
+         ColumnVector result = ParseURI.parseURIQueryWithLiteral(v0, "amount", true)) {
+      String[] expected = {"100", null, null};
+      try (ColumnVector expectedCV = ColumnVector.fromStrings(expected)) {
+        AssertUtils.assertColumnsAreEqual(expectedCV, result);
+      }
+    }
+    
+    // Test that ANSI mode still throws for truly invalid URLs
+    String[] invalidData = {"://completely-malformed"};
+    try (ColumnVector v0 = ColumnVector.fromStrings(invalidData)) {
+      try {
+        ColumnVector result = ParseURI.parseURIQueryWithLiteral(v0, "param", true);
+        result.close();
+        throw new AssertionError("Expected exception for invalid URL in ANSI mode");
+      } catch (RuntimeException e) {
+        assert e instanceof com.nvidia.spark.rapids.jni.ExceptionWithRowIndex;
```

Comment:
Nit, we can use `assertThrows`:

```java
import static org.junit.jupiter.api.Assertions.*;

assertThrows(RuntimeException.class, () -> {
  ParseURI.parseURI......
});
```

Please check all the places which can use `assertThrows`